fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -136,9 +136,16 @@
                     if (limitRemaining != Integer.MAX_VALUE) {
                         limitRemaining--;
                     }
+                    Tuple setPrimaryKey = Tuple.fromBytes(pk.bytes);
                     Tuple tuple = Tuple.fromList(fieldNames);
+                    int[] keyPos = new int[setPrimaryKey.size()];
+                    for (int i = 0; i < setPrimaryKey.size(); i++) {
+                        keyPos[i] = i+fieldNames.size();
+                    }
+                    state.index.setPrimaryKeyComponentPositions(keyPos);
```

Comment:
I think I see how this works, by setting up for a later conversion of this index entry for this index to use the updated positions. But I think the intention was that `getPrimaryKeyComponentPositions` was a fixed piece of the meta-data. Which means that the code here needs to rearrange the order of primary key elements in the tuple, rather than changing the meta-data to match how they are arranged now.

Reply:
So... I tried that and got slightly lost in the logic of trying to match fields that were filled vs fields that weren't and setting the placeholder. I think if I'm correct that its possible to have a larger than size 2 primary key? And that its very possible that the fieldNames don't directly match up with those values that are set since we don't have the __recordID in the field names. I considered this might be something I wanted to give more thought to.

Reply:
I haven't tried myself, but I think if you start with a (mutable) list of field names (as dummies for the indexed text values) and then go through the primary key positions and (if not `-1`) replace the field name _and_ remove from the list to be appended later (or, equivalently, collect just the ones that are `-1`), it'll work.

Reply:
I think the issue is that the field names do not necessarily contain the primary key fields. So it's possible that we will end up with that approach with an inaccurate tuple length or replacing a name that doesn't match the name we're trying to place the value into.

Reply:
Also what is the probability of the field not being in the right place in the list. You know the  overall code better but it seems this relies on both the fieldNames containing the primary key fields that are set being placed in the right position in the field names list. I'm not entirely sure how assured that is.

Reply:
If the primary key field isn't in the declared fields, it isn't in `getPrimaryKeyComponentPositions`, so you don't replace anything and keep it in the list to be appended.